### DISCUSSION TITLE
Advise when file change

### DISCUSSION THREAD

**Discussion by scascalespAV** [microsoft/vscode-discussions#246] [Extension Development QnA] [answered]
Hello

Is there an extension or option for VSCode to notify when the file has been changed from another editor like notepad++ does?

![image](https://user-images.githubusercontent.com/110404394/202225646-d8c5d03f-9686-41a6-b45a-586267fc9843.png)

Thanks

**Comment by ArturoDent** [comment]
Can you create a `FileSystemWatcher` and listen to the `onDidChange/Create/Delete` events?

**Comment by scascalespAV** [comment]
Hello
In extesion? I wouldn't know how to do it

**Comment by gjsjohnmurray** [comment, marked as answer]
If you open a file inside VS Code, then change it externally using another tool, saving the change should already cause a silent reload in VS Code unless you have already made changes to it in VS Code but haven't yet saved them.

**Comment by scascalespAV** [comment]
Yes, but if I edit the file in code, then edit it from outside and save changes from outside, code does not reload or report that the file has changed

Ideally like notepad++: every few seconds code checks the file it has against the original and if it is different (it is being edited or not) notify to reload or not

And it would be ideal if check deleted files outside code:
![image](https://user-images.githubusercontent.com/110404394/202421075-eb23675e-9882-4471-9ecc-d8276c738a06.png)
